File: tests/fixtures/_highlight-corpus/sample.clj
;; Clojure sample
(ns sample.core
  (:require [clojure.java.io :as io]
            [clojure.string :as str])
  (:gen-class))

(defrecord Counter [value])

(defn increment
  ([counter] (increment counter 1))
  ([{:keys [value] :as counter} n]
   (assoc counter :value (+ value n))))

(defn count-bytes [path]
  (with-open [r (io/reader path)]
    (->> (line-seq r)
         (remove str/blank?)
         (reduce (fn [c line] (increment c (count line)))
                 (->Counter 0))
         :value)))

(defn -main [& [path]]
  (let [path  (or path "input.txt")
        total (count-bytes path)]
    (println (format "total bytes: %d" total))))

File: tests/fixtures/_highlight-corpus/sample.pl
#!/usr/bin/env perl
# Perl sample
use strict;
use warnings;
use v5.30;

package Counter {
    sub new { my $class = shift; bless { value => 0 }, $class; }
    sub increment {
        my ($self, $n) = @_;
        $self->{value} += $n // 1;
        return $self->{value};
    }
    sub value { $_[0]->{value} }
}

sub main {
    my ($path) = @_;
    open(my $fh, '<', $path) or die "open $path: $!";
    my $counter = Counter->new();
    while (my $line = <$fh>) {
        chomp $line;
        $counter->increment(length $line) if length $line > 0;
    }
    close($fh);
    printf "total bytes: %d\n", $counter->value;
}

main($ARGV[0] // 'input.txt');
